iodata/overlap.py

```diff
@@ -81,16 +86,47 @@ def compute_overlap(obasis: MolecularBasis, atcoords: np.ndarray) -> np.ndarray:
 
             # START of Cartesian coordinates. Shell types are positive
             result = np.zeros((len(scales[i0][0]), len(scales[i1][0])))
-            # Loop over primitives in shell0 (Cartesian)
-            for iexp0, (a0, cc0) in enumerate(zip(shell0.exponents, shell0.coeffs[:, 0])):
-                s0 = scales[i0][iexp0]
 
-                # Loop over primitives in shell1 (Cartesian)
-                for iexp1, (a1, cc1) in enumerate(zip(shell1.exponents, shell1.coeffs[:, 0])):
-                    s1 = scales[i1][iexp1]
-                    n0 = np.vstack(list(iter_cart_alphabet(shell0.angmoms[0])))
-                    n1 = np.vstack(list(iter_cart_alphabet(shell1.angmoms[0])))
-                    add_overlap(cc0 * cc1, a0, a1, s0, s1, r0, r1, n0, n1, result)
+            a0 = np.min(shell0.exponents)
+            a1 = np.min(shell1.exponents)
+            # arrays of angular momentums [[2, 0, 0], [0, 2, 0], ..., [0, 1, 1]]
+            n0 = np.array(iter_cart_alphabet2(shell0.angmoms[0]))
+            n1 = np.array(iter_cart_alphabet2(shell1.angmoms[0]))
```

Comment:
Can these two go into the `if prefactor > 1.e-15:` block? I don't expect much performance gain. It just makes things easier to follow.

Reply:
For sure, I should have caught that.